src/dashboard.js

```diff
@@ -601,6 +615,34 @@ class Dashboard extends Component {
     window.open(link);
   }
 
+  // type 0 all devices, 1 one device, 2 endpoint
+  showAllStorageTable = (type, id) =>{
+    var table = [];
+    this.records.forEach((value,key)=>{
+      if(type === 0 && value.operationName === 'DiagnosticIoTHubIngress')
+      {
+        table.push(value);
+      }
+      else if (type === 1 && value.operationName === 'DiagnosticIoTHubIngress' && value.properties.deviceId === id)
+      {
+        table.push(value);
+      }
+      else if (type === 2 && value.operationName === 'DiagnosticIoTHubRouting')
+      {
+        table.push(value);
+      }
+    })
+    this.state.storageTable = table;

```

Comment:
Do not use this.state.xxx as a left value. Better to move storageTable out of state because it doesn't cause re-render in current component

Reply:
update: just change to setState is OK